daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/PState.scala

```diff
@@ -197,6 +197,7 @@ final class PState private (
 
   def mark(requestorID: String): PState.Mark = {
     // threadCheck()
+    changedVariablesStack.push(mutable.MutableList[GlobalQName]())
```

Comment:
Please add comment that there is an invariant that every mark must be either reset to, or discarded in order for this stack to be properly maintained.

Reply:
Please add the comment as Mike asked